sbnana/CAFAna/Core/Utilities.cxx

```diff
@@ -613,6 +613,16 @@ namespace ana
 
         assert(!r2.empty());
 
+        // Strip the outermost enclosing []
+        const std::string& r2tmp = r2.substr(1, r2.length()-2);
+
+        // Check that the run/subrun is not already present
+        if (r1.find(r2tmp) != std::string::npos){
+          std::cout << "Found files with duplicate run/subrun metadata: " << r2tmp << std::endl;
+
+          assert(false);
```

Comment:
I am not sure what the best convention to follow here is. Usually I would throw a `art::exception` but I know that this repository wants to avoid any dependence on art. I can use a `std::exception` if that is preferable? Alternatively if there is a way to include a message containing the run/subrun that caused the issue in the assert I am happy to do that.

Reply:
I have a strong dislike of exceptions in most cases arising in physics code. They only make sense if there's some potential sensible way to catch and handle them, which we don't have here. They also have a terrible habit of killing your job with no better message than "uncaught exception", and then they're also a pain to catch in the debugger, because people not as enlightened as us love to throw and catch them as part of the normal control throw [shakes fist at fhicl::ParameterSet].

So I think this is absolutely fine. I would probably change the `assert(false)` to just `abort()`. It's a bit more explicit, and it can't get optimized away if anyone ever changes the assertion settings.

Reply:
FYI, `DuplicateEventTracker` in `icaruscode` provides some functionality like this.